conn_linux.go

```diff
@@ -161,24 +158,16 @@ func (c *conn) Receive() ([]Message, error) {
 	}
 
 	// Read out all available messages
-	n, _, _, from, err := c.s.Recvmsg(b, nil, 0)
+	n, _, _, _, err := c.s.Recvmsg(b, nil, 0)
 	if err != nil {
-		return nil, err
-	}
-
-	addr, ok := from.(*unix.SockaddrNetlink)
-	if !ok {
-		return nil, errInvalidSockaddr
-	}
-	if addr.Family != unix.AF_NETLINK {
-		return nil, errInvalidFamily
+		return nil, os.NewSyscallError("recvmsg", err)
 	}
 
 	n = nlmsgAlign(n)
 
 	raw, err := syscall.ParseNetlinkMessage(b[:n])
 	if err != nil {
-		return nil, err
+		return nil, os.NewSyscallError("syscall-parse-netlink-message", err)
```

Comment:
Yet another pedantic nit, sorry!

This code is calling a function from the `syscall` package, but that function is not actually making any system calls: https://golang.org/src/syscall/netlink_linux.go#L116.

It doesn't seem like this is a condition that the caller can recover from anyway. Perhaps a `fmt.Errorf("failed to parse netlink message: %v", err)` would be more semantically precise.